modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java

```diff
@@ -423,6 +482,57 @@ public void testWalCompressedBytes() throws Exception {
         assertCorrectWalCompressedBytesMetrics(n1);
     }
 
+    /**
+     * populates a given cache w/32 KB of data
+     * @param cacheName  - name of cache to populate
+     * @param igniteEx - handle to ignite
+     */
+    private void populateCache(String cacheName, IgniteEx igniteEx) {
+        for (int i = 0; i < 10; i++)
+            igniteEx.cache(cacheName).put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+    }
+
+    /**
+     * Turns off WAL.
+     * @param igniteEx - handle to ignite
+     * @throws IgniteCheckedException
+     */
+    private void turnOffWal(IgniteEx igniteEx) throws IgniteCheckedException {
+        WALDisableContext walDisableCtx = igniteEx.context().cache().context().walState().walDisableContext();
+        assertNotNull(walDisableCtx);
+
+        setFieldValue(walDisableCtx, "disableWal", true);
+
+        assertTrue(walDisableCtx.check());
+        assertNull(walMgr(igniteEx).log(new DataRecord(emptyList())));
+    }
+
+    /**
+     *  check the state of wal archive, and whether the total size of wal and (possibly) wal archive match what is expected.
+     * @param igniteEx - handle to ignite
+     * @param hasWalArchive - whether wal archiving is enabled
+     * @param addArchiveDirToTotalSize  -- whether the files in the wal archive dir are added tot total size
+     * @throws IgniteCheckedException
+     * @throws InterruptedException
+     */
+    private void checkWalArchiveAndTotalSize(IgniteEx igniteEx, boolean hasWalArchive, boolean addArchiveDirToTotalSize) throws IgniteCheckedException, InterruptedException {
+        FileWriteAheadLogManager walMgr = walMgr(igniteEx);
+
+        SegmentRouter router = walMgr.getSegmentRouter();
+
+        assertEquals(router.hasArchive(), hasWalArchive);
+
+        Thread.sleep(1000); //wait to avoid race condition with FileWriteAheadLogManager.FileArchiver#allocateRemainingFile where wal.tmp files are created.
```

Comment:
This is not correct, it can lead to flaky tests, you can use a condition, for example:
```
        int walSegments = n.context().config().getDataStorageConfiguration().getWalSegments();
        waitForCondition(() -> walMgr(n).walWorkFiles().length != walSegments, getTestTimeout());
        //walWorkFiles - It needs to be implemented like FileWriteAheadLogManager#walArchiveFiles.
```

Reply:
this needs further investigation.